src/tests/system/tests/test_ipa.py

```diff
@@ -640,3 +641,513 @@ def test_ipa__switch_user_with_smartcard_authentication(client: Client, ipa: IPA
+    :setup:
+      1. Add users and create host groups
+      2. Assign a host to a host group
+      3. Disable default HBAC rule
+      4. Create HBAC rule for a host group SSH access
+    :steps:
+      1. Restart SSSD and test SSH login for users
+      2. Remove a host from a host group, restart SSSD and retest SSH login for users
+    :expectedresults:
+      1. Permitted user SSH login succeeds
+      2. Host removed from hostgroup successfully, SSSD restarted; user access denied after host group removal
+    :customerscenario: False
+    """
+    users = ["user1", "user2"]
+    for user in users:
+        ipa.user(user).add()
+    allow_hst_gr = ipa.hostgroup("allow_host_group").add(description="allow servers group")
+    ipa.hostgroup("not_allowed").add(description="not allowed servers group")
+    allow_hst_gr.add_member(host="client.test")
+    ipa.hbac("allow_all").disable()
+    ipa.hbac("allow_host_group_ssh_access").create(
+        description="SSH access for a host group", users="user1", hostgroups="allow_host_group", services="sshd"
+    )
+
+    client.sssd.restart()
+    assert client.auth.ssh.password("user1", "Secret123"), "user1 SSH should succeed!"
+    assert not client.auth.ssh.password("user2", "Secret123"), "user2 SSH should be denied!"
+
+    allow_hst_gr.remove_member(host="client.test")
+    client.sssd.restart()
+    assert not client.auth.ssh.password("user1", "Secret123"), "user1 SSH should be denied after host group removal!"
+
+
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__hbac_service_group_access(client: Client, ipa: IPA):
+    """
+    :title: Validate HBAC rule-based access control using service groups
+    :setup:
+      1. Add users and create a service group with sshd and login services
+      2. Disable default HBAC rule
+      3. Create HBAC rule using the service group
+    :steps:
+      1. Restart SSSD and validate SSH login for users
+      2. Remove sshd from service group; restart SSSD and verify users access
+    :expectedresults:
+      1. SSH login succeeds for a user which is added in rule
+      2. user denied access after removal of a service from service group
+    :customerscenario: False
+    """
+    users = ["user1", "user2"]
+    for user in users:
+        ipa.user(user).add()
+    remote_svc_group = ipa.hbacsvcgroup("remote_access").add(description="Remote access services")
+    remote_svc_group.add_member(hbacsvc=["sshd", "login"])
+    ipa.hbac("allow_all").disable()
+    ipa.hbac("remote_services_access").create(
+        description="Remote access via service groups",
+        users="user1",
+        hosts="client.test",
+        servicegroups="remote_access",
+    )
+
+    client.sssd.restart()
+    assert client.auth.ssh.password("user1", "Secret123"), "user1 SSH should succeed!"
+    assert not client.auth.ssh.password("user2", "Secret123"), "user2 SSH should be denied!"
+
+    remote_svc_group.remove_member(hbacsvc=["sshd"])
```

Comment:
I think this is a little repetitive. for some tests, we want to make sure the latest changes reflect. This is not testing that. Instead of reverting the changes, let's test it before. 

1. All users can login
2. Apply hbac
3. User in hbac works, user not in hbac is denied 


We can actually do more with this, su and sshd are different services. User can SU but cannot SSH. To better test these rules.

Reply:
you mean should I remove `su` instead of `sshd`?